packages/transaction/src/tx-builder.ts

```diff
@@ -318,12 +318,14 @@ export class TxBuilder {
       'inflow != outflow',
     )
     for (const addr of Object.keys(inflowSum.erc20)) {
+      if (outflowSum.getERC20(addr).eq('0')) continue
```

Comment:
to avoid if a UTXO includes eth and erc20 deposit but the UTXO is chosen for eth transfer only.